src/NUnitFramework/framework/Constraints/EqualConstraintResult.cs

```diff
@@ -17,8 +17,9 @@ public class EqualConstraintResult : ConstraintResult
         private readonly Tolerance _tolerance;
         private readonly bool _caseInsensitive;
         private readonly bool _ignoringWhiteSpace;
+        private readonly bool _comparingProperties;
         private readonly bool _clipStrings;
-        private readonly IList<NUnitEqualityComparer.FailurePoint> _failurePoints;
+        private readonly IList<NUnitEqualityComparer.FailurePoint>? _failurePoints;
```

Comment:
Does this need to be null?  It could be just an empty list.  Would also simplify some code below.

Reply:
This change was deliberate. Previously the `NUnitEqualityComparer` would allocate a new list for every `AreEqual` call, regardless if there were ever any failure points. Now it is only allocated when needed.
This should speed up all `AreEqual` calls, especially assuming that 99.9% succeed and only a fraction of the failing ones even have failure points.

However I have updated the code in this class to remove `null` and initialize it with `Array.Empty`.
Now various null checks can be removed. Thanks for that.